components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -81,20 +88,35 @@ public void handle(HttpServletRequest request, HttpServletResponse response, Aut
             log.trace("Inside handle()");
         }
         SequenceConfig sequenceConfig = context.getSequenceConfig();
+        // Retrieve session information from cache.
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(context.getSessionIdentifier());
         ExternalIdPConfig externalIdPConfig = null;
         if (FrameworkServiceDataHolder.getInstance().getAuthnDataPublisherProxy() != null &&
-                FrameworkServiceDataHolder.getInstance().getAuthnDataPublisherProxy().isEnabled(context)) {
-            // Retrieve session information from cache in order to publish event
-            SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(context.getSessionIdentifier());
-            if (sessionContext != null) {
-                Object authenticatedUserObj = sessionContext.getProperty(FrameworkConstants.AUTHENTICATED_USER);
-                AuthenticatedUser authenticatedUser = new AuthenticatedUser();
-                if (authenticatedUserObj != null) {
-                    authenticatedUser = (AuthenticatedUser) authenticatedUserObj;
+                FrameworkServiceDataHolder.getInstance().getAuthnDataPublisherProxy().isEnabled(context) &&
+                    sessionContext != null) {
+            Object authenticatedUserObj = sessionContext.getProperty(FrameworkConstants.AUTHENTICATED_USER);
+            AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+            if (authenticatedUserObj != null) {
+                authenticatedUser = (AuthenticatedUser) authenticatedUserObj;
+            }
+            FrameworkUtils.publishSessionEvent(context.getSessionIdentifier(), request, context,
+                    sessionContext, authenticatedUser, FrameworkConstants.AnalyticsAttributes
+                            .SESSION_TERMINATE);
+        }
+
+        // Remove federated authentication session details from the database.
+        if (sessionContext != null && StringUtils.isNotBlank(context.getSessionIdentifier()) &&
+                sessionContext.getSessionAuthHistory().getHistory() != null) {
```

Comment:
Can't the `sessionContext.getSessionAuthHistory()` e null? If so do the null check?

Reply:
Fixed with PR https://github.com/wso2/carbon-identity-framework/pull/3738